### DISCUSSION TITLE
gotox今天也不能用了

### DISCUSSION THREAD

**Discussion by caofengzi** [SeaHOH/GotoX#51] [点子]
吉林聯通，今天gotox也不能使用了，應該可用ip都沒有了。

**Comment by SeaHOH** [comment]
是的。
可以找一些 sniproxy ip 来使用，一定要作为固定列表（`[gae/iplist]`），不然可能会影响其他人使用。
或者换用其它还能使用的代理。

**Comment by SeaHOH** [comment]
用 sniproxy ip 的话，servername 一定要是谷歌域名，但不能是 `www.google.com`。

**Comment by caofengzi** [comment]
這次是針對什麼封鎖的？ip還是證書。

**Comment by dou4cc** [comment]
ip+证书+servername

**Comment by Just2co** [comment]
QUIC目前依然坚挺，teredo IPv6隧道依然高速可用，sock5前置代理依旧如常高速可用，目前可以利用这3种方式高速过墙！

**Comment by tec1987** [comment]
@Just2co
teredo IPv6用的哪家的？以前用过，感觉速度一般

**Comment by Just2co** [comment]
@tec1987 韩国那家，虽然无法ping通，但是填进去重启PC之后可以用！速度还行，没以前快但是胜在稳定1080P看油管。

**Comment by masktao** [comment]
@Just2co 请问你说的填入重启pc是什么原理？为啥要重启？goproxy也可以这样用吗？

**Comment by Just2co** [comment]
@masktao 也可以的，不过要自己找IPv6 IP地址放进去，然后再配置文件里启用IPv6支持就可！

**Comment by blob2015** [comment]
额外提一句，上yt、g之类的，都没啥大问题，但上gitter.im，一片的黄红，用quic的话
gp还是比较低调，栏deng、ss是第一线

**Comment by Just2co** [comment]
@SeaHOH 海兄，啥时候升级支持http前置代理啊？我看配置文件前置代理设置项不可用啊，我这里有好几个高速稳定的sock5前置代理，转成http之后，可以作为gotoX的前置代理！

**Comment by SeaHOH** [comment]
好的，我这几天试着添加一下。

**Comment by Just2co** [comment]
@SeaHOH 之所以请求你的帮助，是因为谷歌证书还没被GFW block之前，你写的GOTOX速度最快，放几个IP进去都能跑满带宽，而xx-net速度就不行，goagent在chorome升级到最新版之后证书错误也不能用了。我现在用的xx-net就是开了http前置，它能支持sock4/sock5/http这三种前置。如果你的GotoX也支持了前置，那之前搜的一些高速IP就能继续用上了！

**Comment by SeaHOH** [comment]
那是因为用的 IP 本来就快，GotoX 挑着快的用。
如果有高质量前置，使用固定列表替换自动筛选会比较好，只需要放几个 IP 就行。

**Comment by SeaHOH** [comment]
@Just2co 
已添加，但是未测试，如果不能正常使用，请新开 issues 报错。
>如果有高质量前置，使用固定列表替换自动筛选会比较好，只需要放几个 IP 就行。

放域名应该也可以，只要解析出的 IP 是前置能访问的（多半是台湾 IP），但是这样不保证速度。

**Comment by tec1987** [comment]
@SeaHOH
老兄，现在sniproxy 好像没几个可用的了吧？
GoP只能开QUIC用了。。。
你这个gotoX是基于goa的吧？

**Comment by SeaHOH** [comment]
>现在sniproxy 好像没几个可用的了吧？

可以自己搭，sniproxy 还是比较可靠的只要不连谷歌、脸书等。
连 GAE 时可以用 GoProxy 或 GotoX 自定义 servername，以避开某些关键字。

>GoP只能开QUIC用了

大概是，用 sniproxy 的情况上面写了。

>你这个gotoX是基于goa的吧？

是，但调整了代理流程，细节也有改变。具体可以看 readme 或源码。

**Comment by SeaHOH** [comment]
proxylist 填好没有？

**Comment by SeaHOH** [comment]
没懂，什么意思？你要填进去 GotoX 才能使用。

**Comment by SeaHOH** [comment]
这个只是 GAE 前置，不是 GotoX 前置，GotoX 前置可以使用其它软件实现。

**Comment by SeaHOH** [comment]
的确有 bug，等会儿改……

**Comment by Just2co** [comment]
因为谷歌证书被GFW认证，因此gotox单靠GWS/GVS IP跟GAE沟通已经行不通了，因此我找了MEOW前置SOCKS/http代理作为跳板，然而gotox设置好前置后，并没有起作用,没有数据到MEOW这里来。

**Comment by SeaHOH** [comment]
@Just2co 
已修复 bug，其中一个是 SOCKS4 的旧 bug。
但是你无法使用似乎不光是 bug 的原因，你需要在 proxylist 中填写代理。

**Comment by SeaHOH** [comment]
刚试了好几个，都无法链接 GAE，换了 servername 也不行，似乎针对 SOCKS 代理检查了证书。

**Comment by SeaHOH** [comment]
好像又不是证书，情况比较奇怪，没有条件懒得研究了。

前置应该是废了，GFW 太厉害，SOCKS 太废……

**Comment by Just2co** [comment]
@SeaHOH 白天的时候干扰比较轻，可以再测试一下，找美国的sock5代理，谷歌一大把，只要能ping通就能用，不过最好是用MEOW转换成http，然后再填入前置代理，我前置代理xx-net/goproxy/轮子的两款/蓝灯/赛风都高速稳定可用！

**Comment by SeaHOH** [comment]
上面这个错误，是因为你没有填写链接 GAE 使用的列表。
~默认列表是 google_gws，可以在 [gae/iplist] 修改要使用的列表，然后在指定列表中填域名或几个 IP。
使用前置时，最好不要使用默认列表，可以随便换个名称，方便自定义 IP。~

另外，我这里前置完全无法连接 GAE，http 和 socks 都是如此。

**Comment by SeaHOH** [comment]
使用 GAE 前置时，增加了一个专用列表 `google_2p`，填写谷歌域名或 IP 就行，也可以自定义列表名称。

**Comment by SeaHOH** [comment]
不用给我，我这里标准代理无法突破。

你上面使用了 `g` 作为列表名称，需要在 [iplist] 小节配置
```ini
[iplist]
g = www.google.com | ip1 | ip2
```

**Comment by SeaHOH** [comment]
你只填了一个代理，我没考虑到这个情况，现已修复。

**Comment by SeaHOH** [comment]
使用 GAE 前置时记得修改默认的谷歌直连规则，封锁严重时这是连不上的，配置成前置使用的代理或其它代理。

**Comment by Just2co** [comment]
@SeaHOH 我在iplist这里填了好几个IP，只不过只给你复制了这一个，问题是MEOW作为gotox的前置的时候，MEOW这里收不到gotox过来的数据包，不知道你修复的是不是这个问题

**Comment by SeaHOH** [comment]
>File "D:\Program Files\Goproxy\GotoX\local\HTTPUtil.py", line 569, in get_gae_front
return random.choice((proxy_list[0], random.choice(proxy_list[1:])))
File "random.py", line 264, in choice
IndexError: Cannot choose from an empty sequence

我是从这里看出来的，你只填了一个代理，不是指 IP。
代码错误后就不会发出请求，所以 MEOW 没有收到数据包，现已修复。

**Comment by SeaHOH** [comment]
其实你只需要配置 `gae` 小节的 `enableproxy` 和 `proxylist` 就行了。

如果无法正确解析 `www.google.com`，就把 `iplist` 小节的 google_2p 配置为几个 IP。

**Comment by SeaHOH** [comment]
已修改。加油，快成功了~

**Comment by SeaHOH** [comment]
已修改，我觉得这回行了。

socks 由 1.6.6 升级到 1.6.7，要删掉旧的 `PySocks-1.6.6.egg`。

再提醒下，使用 GAE 前置时记得修改默认的谷歌直连规则为代理规则。
明天想办法改一下这个流程，默认谷歌规则直接在内部替换成走 GAE 前置，不用再自行修改。

**Comment by SeaHOH** [comment]
>不用给我，我这里标准代理无法突破。

前言撤回~ 如果有自建的给我个作为后备，发我邮箱就行。 :smile_cat: 

刚试了一下，居然成功使用前置突破。看来我还是太想当然了，没有实际运行验证。
检查了之前的测试代码才发现，之前用的默认 ssl 模块，都是默认设置（可能是这个部分参数不安全导致被墙） 。

**Comment by SeaHOH** [comment]
@Just2co 
我已经能用了，怀疑配置错误。你的配置发我邮箱，明晚答复你。

今天累死了，明天也比较忙，我要 :bed: :zzz: 了。

**Comment by 2378** [comment]
mark一下 是时候多备几个梯子了

**Comment by tec1987** [comment]
@Just2co 肯定不是，自己看域名后缀。。。

**Comment by SeaHOH** [comment]
@Just2co 
下面这个是 socks5 吗？
```ini
proxylist = HTTP://127.0.0.1:4411
```
正确写法是
```ini
proxylist = socks5://127.0.0.1:4411
```

**Comment by SeaHOH** [comment]
好像不是这个原因，你上面实际上已经成功链接代理了，但是返回的结果不对，导致进行了有效性测试。

换个刚刚在 XX-Net 成功使用的代理试试？

**Comment by SeaHOH** [comment]
这里面有几个脸书的 IP？
69.171.229.73|74.125.23.138|74.125.23.139|31.13.65.1|31.13.65.17|69.171.235.101|74.125.23.101|74.125.23.113|74.125.23.102

**Comment by SeaHOH** [comment]
我这儿今天又用不了前置了……

**Comment by SeaHOH** [comment]
刚修改成不用手动更改默认谷歌直连规则，可自动使用 GAE 的前置代理。

**Comment by SeaHOH** [comment]
@Just2co
今天测试 XX-Net 的时候，顺便也用 X-Tunnel 测试了一下 GotoX 的 GAE 前置。
X-Tunnel 是加密传输的，结果可用于 GotoX 前置，同时普通标准代理仍然无法连接，前置不了。

**Comment by Just2co** [comment]
@SeaHOH 估计还是哪里写的有问题，同样的前置，用XX-NET的正常，就是GOTOX就不行。

**Comment by SeaHOH** [comment]
我使用普通代理前置 XX-Net 还是无法使用。
你确定 XX-Net 走了前置代理吗？如果是，我就放弃了，没法测试。
